Port LoadParameters to the new IPC serialization format

cdumez opened the pull request:
#### f739f0316b1e91338abd95a21d2f98c66ab7207a
<pre>
Port LoadParameters to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=264841">https://bugs.webkit.org/show_bug.cgi?id=264841</a>

Reviewed by Alex Christensen.

Port LoadParameters to the new IPC serialization format. Add a
[EncodeRequestBody] member attribute to support serialization /
deserialization of the ResourceRequest body since this is the
second time we do such a thing.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(encode_type):
(decode_type):
* Source/WebKit/Shared/Cocoa/LoadParametersCocoa.mm:
(WebKit::LoadParameters::platformEncode const): Deleted.
(WebKit::LoadParameters::platformDecode): Deleted.
* Source/WebKit/Shared/LoadParameters.cpp: Removed.
* Source/WebKit/Shared/LoadParameters.h:
* Source/WebKit/Shared/LoadParameters.serialization.in: Added.
* Source/WebKit/Shared/URLSchemeTaskParameters.cpp: Removed.
* Source/WebKit/Shared/URLSchemeTaskParameters.h:
* Source/WebKit/Shared/URLSchemeTaskParameters.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/270762@main">https://commits.webkit.org/270762@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8b6c2b2b020c1083aab443f0eda4e65d3c53c595

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26303 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4912 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27561 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28398 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24072 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6697 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2323 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24068 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26559 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3755 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22613 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28977 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/26466 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3347 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29645 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23969 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23979 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27536 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3385 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1588 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4809 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6328 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3862 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3709 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (approved):
Maybe, to make it easier to see where functionality would change, we could make an attribute to encode the body and an attribute to omit the body and ResourceRequest would always have to have one or the other.  This is fine as-is

cdumez committed:
Port LoadParameters to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=264841

Reviewed by Alex Christensen.

Port LoadParameters to the new IPC serialization format. Add a
[EncodeRequestBody] member attribute to support serialization /
deserialization of the ResourceRequest body since this is the
second time we do such a thing.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(encode_type):
(decode_type):
* Source/WebKit/Shared/Cocoa/LoadParametersCocoa.mm:
(WebKit::LoadParameters::platformEncode const): Deleted.
(WebKit::LoadParameters::platformDecode): Deleted.
* Source/WebKit/Shared/LoadParameters.cpp: Removed.
* Source/WebKit/Shared/LoadParameters.h:
* Source/WebKit/Shared/LoadParameters.serialization.in: Added.
* Source/WebKit/Shared/URLSchemeTaskParameters.cpp: Removed.
* Source/WebKit/Shared/URLSchemeTaskParameters.h:
* Source/WebKit/Shared/URLSchemeTaskParameters.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/270762@main
